From c65a606ae73c9f9fa7bed4aade575395f4ff1890 Mon Sep 17 00:00:00 2001
From: David Henningsson <david.henningsson@canonical.com>
Date: Mon, 30 Mar 2015 11:10:56 +0200
Subject: [PATCH 309/311] resampler: Allow disabling the LFE filter by setting
 crossover to 0

When crossover_freq is set to 0, this restores the old behaviour
of letting the LFE channel be the average of the source channels,
without additional processing. This can be useful e g in case the
user already has a hardware crossover.

Signed-off-by: David Henningsson <david.henningsson@canonical.com>
---
 man/pulse-daemon.conf.5.xml.in | 2 +-
 src/pulsecore/resampler.c      | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

Index: pulseaudio/man/pulse-daemon.conf.5.xml.in
===================================================================
--- pulseaudio.orig/man/pulse-daemon.conf.5.xml.in	2015-05-13 15:00:32.082148312 +0800
+++ pulseaudio/man/pulse-daemon.conf.5.xml.in	2015-05-13 15:00:32.078148312 +0800
@@ -122,7 +122,7 @@
 
     <option>
       <p><opt>lfe-crossover-freq=</opt> The crossover frequency (in Hz) for the
-      LFE filter. Defaults to 120 Hz.</p>
+      LFE filter. Defaults to 120 Hz. Set it to 0 to disable the LFE filter.</p>
     </option>
 
     <option>
Index: pulseaudio/src/pulsecore/resampler.c
===================================================================
--- pulseaudio.orig/src/pulsecore/resampler.c	2015-05-13 15:00:32.082148312 +0800
+++ pulseaudio/src/pulsecore/resampler.c	2015-05-13 15:00:32.078148312 +0800
@@ -394,7 +394,7 @@
     if (r->map_required)
         setup_remap(r, &r->remap, &lfe_filter_required);
 
-    if (lfe_filter_required) {
+    if (lfe_filter_required && crossover_freq > 0) {
         pa_sample_spec wss = r->o_ss;
         wss.format = r->work_format;
         /* FIXME: For now just hardcode maxrewind to 3 seconds */
